cli/command/service/opts.go

```diff
@@ -797,6 +798,7 @@ func addServiceFlags(flags *pflag.FlagSet, opts *serviceOptions, defaultFlagValu
 	flags.StringVar(&opts.endpoint.mode, flagEndpointMode, defaultFlagValues.getString(flagEndpointMode), "Endpoint mode (vip or dnsrr)")
 
 	flags.BoolVar(&opts.registryAuth, flagRegistryAuth, false, "Send registry authentication details to swarm agents")
+	flags.BoolVar(&opts.noResolveImage, flagNoResolveImage, false, "Do not query the registry to resolve image digest and supported platforms")
```

Comment:
Do we really want to hide this flag when talking to an older daemon? I'd think you would want to use the flag when the daemon is old, because the daemon wouldn't support the "distribution inspect" endpoint that gets used without this flag.